File: Java/045.跳跃游戏II.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        if(n <= 1) return 0;
        int index = 0, curMax = 0, step = 0, nextMax = 0;
        while(index <= curMax){
            while(index <= curMax){
                nextMax = Math.max(nextMax, index + nums[index]);
                index++;
            }
            curMax = nextMax;
            step++;
            if(curMax >= nums.length - 1) return step;
        }
        return 0;
    }
}

File: CPP/045. 跳跃游戏 II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int maxReach = 0, jump = 0, curReach = 0;
        for(int i =0; i < n; i++){
            if(maxReach >= n - 1) return jump;
            if(maxReach < i){
                jump++;
                maxReach = curReach;
            }
            curReach = max(curReach, nums[i] + i);
        }
        return jump;
    }
};